search/ranker.cpp

```diff
@@ -46,12 +48,93 @@ void RemoveDuplicatingLinear(vector<IndexedValue> & values)
                       }),
                values.end());
 }
+
+void RemoveStringPrefix(string const & str, string & res)
+{
+  search::Delimiters delims;
+  // Find start iterator of prefix in input query.
+  using TIter = utf8::unchecked::iterator<string::const_iterator>;
+  TIter iter(str.end());
+  while (iter.base() != str.begin())
+  {
+    TIter prev = iter;
+    --prev;
+
+    if (delims(*prev))
+      break;
+    else
+      iter = prev;
+  }
+
+  // Assign result with input string without prefix.
+  res.assign(str.begin(), iter.base());
+}
+
+ftypes::Type GetLocalityIndex(feature::TypesHolder const & types)
+{
+  using namespace ftypes;
+
+  // Inner logic of SearchAddress expects COUNTRY, STATE and CITY only.
+  Type const type = IsLocalityChecker::Instance().GetType(types);
+  switch (type)
+  {
+  case NONE:
+  case COUNTRY:
+  case STATE:
+  case CITY: return type;
+  case TOWN: return CITY;
+  case VILLAGE: return NONE;
+  case LOCALITY_COUNT: return type;
+  }
+}
+
+/// Makes continuous range for tokens and prefix.
+template <class TIter, class ValueT>
+class CombinedIter
```

Comment:
Is it possible to get rid of CombinedIter (as it's used in a quite specific case) and use StringSlice instead?